src/main/java/com/selfxdsd/selfweb/api/input/validators/RoleValidator.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2020, Self XDSD Contributors
+ * All rights reserved.
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to read the Software only. Permission is hereby NOT GRANTED to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software.
+ * <p>
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.selfweb.api.input.validators;
+
+import com.selfxdsd.api.Contract;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Arrays;
+
+/**
+ * Implementation of a role validator. It checks
+ * if input matches existing {@link Contract.Roles}.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.1
+ */
+public final class RoleValidator implements
+    ConstraintValidator<ValidRole, String> {
+
+    @Override
+    public boolean isValid(final String role,
+                           final ConstraintValidatorContext context) {
+        return Arrays
+            .stream(Contract.Roles.class.getDeclaredFields())
+            .map(field -> {
+                try {
+                    return (String) field.get(null);
+                } catch (final IllegalAccessException exception) {
+                    throw new RuntimeException(exception);
+                }
+            })
+            .anyMatch(role::equalsIgnoreCase);
+    }
```

Comment:
@criske about unit testing: it's enough if you just write some unit tests for this class.